luomu-common/src/macaddr.rs

```diff
@@ -1,61 +1,94 @@
-use std::convert::TryFrom;
+use std::convert::{TryFrom, TryInto};
 use std::fmt;
 use std::str::FromStr;
 
+use crate::TagError;
+
 use super::InvalidAddress;
 
 /// A Mac address used for example with Ethernet.
 ///
 /// Mac address is handled as big endian value. All `From<T>` implementations
-/// returning `MacAddr` expect input as big endian. `From<u64>` also expects
+/// returning [MacAddr] expect input as big endian. `From<u64>` also expects
 /// address to reside in lowest 6 bytes. All `From<MacAddr>` and
 /// `TryFrom<MacAddr>` implementations return their bytes as big endian.
+///
+/// The [MacAddr] type has space to store one VLAN ID (a tag). See
+/// [crate::TaggedMacAddr] for type with more space for storing tag stack.
 #[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
The ordering of `MacAddress` is now a bit different as the tag is in the most significant portion of the u64. Two mac addresses are sorted differently if they have vlan tag set. Also two mac addresses are not equal even if they have the same address bytes but they contain different vlan information.

There is currently no way to check if two addresses are the same (that is, they contain same address bytes) without also a) ensuring there is no vlan tag in neither or b) popping tags from both. 

I am not sure this matters, but to me this makes the ergonomics worse.

Reply:
Sorting is now done by mac address first and then by vlan tag.

For checking only mac address there's now `eq_mac()` method and to check vlan tag there's `eq_tag()` method. `PartialEq` and `Eq` check both.